### ISSUE TITLE
Adding package to Xcode project.

### ISSUE THREAD

**Issue by bhartsb** [liuliu/swift-mujoco#3]
First thank you for this great project!

I've created a MacOS command line tool app in Xcode and attempting to add the swift-mujoco package via Xcode:

<img width="645" alt="choosePackage" src="https://user-images.githubusercontent.com/7194775/191076517-f9120216-2b1c-40b0-b165-68c2eeb942f6.png">

After doing this and attempting to build I'm seeing:

<img width="522" alt=" Swift package product 'C_mujoco is linked as a static library by" src="https://user-images.githubusercontent.com/7194775/191076741-3d2f7bba-6084-4426-800d-b71ec3f0ff03.png">

My longer term hope is to be able to show mujoco in a MTKView or SCNView.  I read that NSOpenGLView is deprecated, but maybe it could be used as an interim step to porting Mujoco's OpenGL code.

**Comment by liuliu** [comment]
You need to install glfw locally for the GLContext object dependency.

I would also recommend you first to try the `swift build --target simulate` to see if it works locally before adding as a target to Xcode.

To install glfw on macOS, `brew install glfw`

If you want to replace with NSOpenGLView, you basically need to write your own GLContext, at that point, I recommend fork / edit and then put up a PR to introduce the new GLXXXContext back.

**Comment by bhartsb** [comment]
I had previously installed glfw 3.3.8.  
I can build simulate.  If I try to run it in Terminal I get:
main.swift:3:10: error: no such module 'MuJoCo'
Probably not related to the linking error.  
I'm not clear on the steps to adding simulate as a target.
Simulate is added as a framework when I added the MujoCo package:
<img width="813" alt="image" src="https://user-images.githubusercontent.com/7194775/191124392-73227c1b-43ba-4b74-971f-02e88690ae5c.png">

**Comment by liuliu** [comment]
simulate and codegen are not frameworks, but a binary target (one is the same as samples/simulate.cc from main MuJoCo repo, the other is the code generator).

Can you copy the full command line errors? Make sure you run the command from swift-mujoco repo:
```
git clone https://github.com/liuliu/swift-mujoco.git
cd swift-mujoco
swift build --target simulate
```

**Comment by bhartsb** [comment]
Yes I see .o files and swift modules.  Here is the output from building simulate.  Warnings not errors.  The only error is the no such module.
[buildMujoSwiftOutput.txt](https://github.com/liuliu/swift-mujoco/files/9604710/buildMujoSwiftOutput.txt)

**Comment by liuliu** [comment]
If
```sh
swift build --target simulate
```
works. You should be able to find the binary under `.build/debug/simulate`. Alternatively, you should be able to run the simulate app by:
```sh
swift run simulate
```
If all these works, then the library itself is good on your macOS. There are probably just some setup issues on the Xcode end if it fails for your project.

I am trying to teething out your macOS dependency issues v.s. the Xcode project setup issues. If `swift run simulate` shows the simulate app window properly, we can rule out former and focus on what happens to the latter.

**Comment by bhartsb** [comment]
So swift run simulate works, but there are some issues that the UI's left panel is very small.  Resizing GLContext from 1280X720 to 1080p did not help as the panel doesn't get bigger.  I encountered multiple issues so I will give the screenshots to illustrate:

1080p left panel is still too small since it doesn't resize:
<img width="1279" alt="SwiftMujoco 1080p" src="https://user-images.githubusercontent.com/7194775/191729221-78ce916c-08e6-4b2e-9201-0c33beea81d9.png">

Dragging to resize causes display issues:
<img width="1364" alt="SwiftMujoco drag resize" src="https://user-images.githubusercontent.com/7194775/191729999-7ce00d52-e718-441d-a9b5-275249eb5a10.png">

fullscreen looks good, but can't exit fullscreen:
<img width="1188" alt="SwiftMujoco fullscreen" src="https://user-images.githubusercontent.com/7194775/191730150-b7b8f3a6-8d6e-425c-817f-eb30ba037c1a.png">

After killing simulate the desktop drawing is broken.  Just showing the top left corner of my desktop in this screenshot:
<img width="615" alt="SwiftMujoco alters desktop drawing" src="https://user-images.githubusercontent.com/7194775/191730380-ad8d3dea-c1f5-4438-a778-5be8a556bc63.png">

I'm starting to get a slightly better feel for how the project is organized.  
I hadn't initially realized that the simulate target main.swift file in /Examples/simulate was specified in Packages.swift.  
Also I'm not familiar with Bazel, only a tiny familiar with SPM, so figuring out what was happening with swift build --target simulate and swift run simulate also was a bit challenging.  I didn't realize that the "simulate" unix executable wasn't created until the "swift run simulate" step.  There are a lot of files generated in the .build folder.  What helped me sort it out was to use FileMerge to diff between the cloned swift-mujoco folder, the folder post build, and the folder post run.  Also looking at the swift symbolic links:  ls -lsa /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/

<img width="752" alt="Swift symbolic links" src="https://user-images.githubusercontent.com/7194775/191735285-b4e44796-452a-4e6d-920f-98be9fa7049c.png">

 I just added this extra information in case it helps someone else who is new to this.  I'm sure that you are already aware of the symbolic linking to swift-frontend.

**Comment by liuliu** [comment]
Yeah. Bazel is not required to know if you don't develop swift-mujoco library yourself. SwiftPM is required and recommended way to introduce swift-mujoco as a dependency for your project.

As for the UI, try to drag & drop a model into it and see what happens (./Examples/assets/).

**Comment by bhartsb** [comment]
Tried the ant asset.  It is not working.  I compared it to the non swift Mujoco and definitely a lot of features aren't working.  The Fullscreen and resizing as mentioned is broken.  When in fullscreen swiping gets one out of FS as does touching some of the menus 
<img width="220" alt="image" src="https://user-images.githubusercontent.com/7194775/191869517-e002413f-b906-442a-aefa-adc76efcea18.png">. Which don't have anything to do with exiting Fullscreen so gestures are mixed up.
![MujoCoAnt](https://user-images.githubusercontent.com/7194775/191869610-fca83182-ffaf-4254-8a9a-80a3f2afda1c.png)
Per the above the skeleton is not recalculating skinning as it does in the C version of Mujoco.
I haven't really explored the C version of MujoCo yet so it is challenging to make comparisons.

Here is the C version with the ant dropped in:
<img width="1180" alt="CVersionMujoCoAnt" src="https://user-images.githubusercontent.com/7194775/191870151-7000a366-1f8f-4b47-a9b3-900a02b66813.png">

**Comment by liuliu** [comment]
OK. This looks like simply the event loop is not running, hence no event processed. Could be a new issue with Xcode 14 (which I haven't tried) since we use some of Swift Concurrency primitives. What's your environment? (Which Xcode version, what kind of Mac (M1 / M2 / Intel?)).

**Comment by liuliu** [comment]
Just tried with Xcode 14.0.1, on a Intel-based Mac Mini, cannot reprod this issue.

**Comment by bhartsb** [comment]
MacOS BigSur 11.6.7
  swift-driver version: 1.26.21 Apple Swift version 5.5.2 (swiftlang-1300.0.47.5 clang-1300.0.29.30)
  Xcode Version 13.2.1
  Hardware:
  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro16,1
  Processor Name:	8-Core Intel Core i9
  Processor Speed:	2.3 GHz
  Number of Processors:	1
  Total Number of Cores:	8
  L2 Cache (per Core):	256 KB
  L3 Cache:	16 MB
  Hyper-Threading Technology:	Enabled
  Memory:	64 GB

**Comment by liuliu** [comment]
Is it possible for you to upgrade? I don't plan to support 5.5 (5.6.x and above, yes) and it would be easier for me to verify (also BigSur is a bit dicey, as I am not sure how compatible some of the concurrency primitives are).

**Comment by bhartsb** [comment]
I'm not sure.  I think I have a dependency related to Maya 2019.  I will have to carefully think about it.  I have another MacBook that I just got back but needs a complete OS reinstall so either I can try that route or do a careful upgrade.  Either way it will take some time.

**Comment by bhartsb** [comment]
I upgraded to Monterey, Xcode 14 and Swift 5.7.  Things are working better but there are still issues.  It doesn't work with dual monitors or fullscreen mode.   With dual monitors I had to disconnect my external one and then it looks okay on my laptop screen.  If I have external monitor attached then the UI is not legible on either screen.

**Comment by liuliu** [comment]
Right. I didn't optimize for dual monitor or fullscreen. These should act the same as `simulate.cc`
